### PR TITLE
Mw/port dans changes from prepare 1 4 dev (#3104)

### PR DESCRIPTION
**Pull request by wilkermichael**
* build: remove last set-output from yaml

* add patch version for dataplane, now required.

---------

Changes proposed in this PR:
-
-

How I've tested this PR:

How I expect reviewers to test this PR:

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -320,6 +333,17 @@ jobs:
         run:
           cp LICENSE ./control-plane
       - uses: hashicorp/actions-docker-build@v1
```
Comment (DanStough) [inline .github/workflows/build.yml:335]: Is this step empty now?

Comment (wilkermichael) [reply .github/workflows/build.yml:335]: I guess, weird, this wasn't a change I made

Comment (wilkermichael) [reply .github/workflows/build.yml:335]: oh maybe I did... one sec

Comment (DanStough) [reply .github/workflows/build.yml:335]: But a new step with that action was added after the fact. Probably, there was a conflict here.

Comment (wilkermichael) [reply .github/workflows/build.yml:335]: yeah, I didn't get a conflict, I guess this is how it merged the cherry pick

Comment (wilkermichael) [reply .github/workflows/build.yml:335]: Ok, I think I just eliminate this uses, and the docker build gets done down lower

Comment (wilkermichael) [reply .github/workflows/build.yml]: that was a weird diff 😅

Comment (wilkermichael) [reply .github/workflows/build.yml]: @DanStough should be fixed

**Review by DanStough — APPROVED** [review]

**Pull request merged**